File: app/src/main/java/com/chiron/playpalyer/media/extractor/AudioExtractor.java
package com.chiron.playpalyer.media.extractor;

import android.media.MediaFormat;

import com.chiron.playpalyer.media.IExtractor;
import com.chiron.playpalyer.media.MediaFormatCallback;

import java.nio.ByteBuffer;

public class AudioExtractor implements IExtractor {
    private MyMediaExtractor myMediaExtractor;
    public AudioExtractor(String path){
        myMediaExtractor = new MyMediaExtractor(path,MediaType.AUDIO);
    }

    @Override
    public MediaFormat getFormat() {
        return myMediaExtractor.getFormat();
    }

    @Override
    public void getFormat(MediaFormatCallback callback) {
        myMediaExtractor.getFormat(callback);
    }

    @Override
    public int readBuffer(ByteBuffer byteBuffer) {
        myMediaExtractor.setTrack(myMediaExtractor.getmAudioTrack());
        return myMediaExtractor.readBuffer(byteBuffer);
    }

    @Override
    public void setTrack(int track) {
        myMediaExtractor.setTrack(track);
    }

    @Override
    public long seek(long pos) {
        return myMediaExtractor.seek(pos);
    }

    @Override
    public void stop() {
        myMediaExtractor.stop();
    }

    @Override
    public long getCurrentTimeStamp() {
        return myMediaExtractor.getCurrentTimeStamp();
    }

    public int getSampleFlag(){
        return myMediaExtractor.getmCurSampleFlag();
    }
}

File: app/src/main/java/com/chiron/playpalyer/mediac/extractor/AudioExtractor.kt
package com.chiron.playpalyer.mediac.extractor

import android.media.MediaFormat
import com.chiron.playpalyer.mediac.IExtractor
import com.chiron.playpalyer.mediac.MMExtractor
import java.nio.ByteBuffer

/**
 * 音频提取器
 */
class AudioExtractor(path: String) : IExtractor {
    private val mMediaExtractor = MMExtractor(path)
    override fun getFormat(): MediaFormat? {
        return mMediaExtractor.getAudioFormat()
    }

    override fun readBuffer(byteBuffer: ByteBuffer): Int {
        return mMediaExtractor.readBuffer(byteBuffer)
    }

    override fun getCurrentTimeStamp(): Long {
        return mMediaExtractor.getCurrentTimeStamp()
    }

    override fun getSampleFlag(): Int {
        return mMediaExtractor.getSampleFlag()
    }

    override fun seek(pos: Long): Long {
        return mMediaExtractor.seek(pos)
    }

    override fun setStartPos(pos: Long) {
        mMediaExtractor.setStartPos(pos)
    }

    override fun stop() {
        mMediaExtractor.stop()
    }
}